Classes/Plugin/Collection.php

```diff
@@ -403,7 +403,8 @@ protected function showSingleCollection($id)
                     'p' => []
                 ];
             } else {
-                $subparts[$resArray['partof']][$resArray['volume_sorting']] = [
+                // volume_sorting should be always set - but it's not a required field. We append the uid to the array key to make it always unique.
+                $subparts[$resArray['partof']][$resArray['volume_sorting'] . $resArray['uid']] = [
```

Comment:
Won't this lead to wrong sorting if UIDs differ in number of digits? E. g. one volume has `volume_sorting = 1900` and `uid = 10`. Another volume has `volume_sorting = 1901` and `uid = 9`. The intending sorting order obviously would be "1900 before 1901", but because of the appending they actually would be sorted "**1901**9 before **1900**10".

Reply:
```suggestion
                $subparts[$resArray['partof']][$resArray['volume_sorting'] . str_pad($resArray['uid'], 9, "0", STR_PAD_LEFT)] = [
```

Reply:
You're right. This can break the sorting depending on the amount of decimals. Your suggested patch is working. Thank you!